sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/HomeScreen.kt

```diff
@@ -44,12 +47,28 @@ fun HomeScreen(
     )
 }
 
+@OptIn(ExperimentalPermissionsApi::class)
 @Composable
 private fun HomeScreenContent(
     uiModels: List<UiModel>,
     showLoading: IsLoading,
     onItemClick: (UiModel) -> Unit
 ) {
+    val cameraPermissionState = rememberPermissionState(CAMERA)
+    if (cameraPermissionState.status.isGranted) {
+        Timber.d("${cameraPermissionState.permission} granted")
+    } else {
+        if (cameraPermissionState.status.shouldShowRationale) {
+            Timber.d("${cameraPermissionState.permission} needs rationale")
+        } else {
+            Timber.d("Request cancelled, missing permissions in manifest or denied permanently")
+        }
+
+        LaunchedEffect(Unit) {
+            cameraPermissionState.launchPermissionRequest()
+        }
+    }
```

Comment:
@luongvo I think we should move the section of handing the `state` to `HomeScreen` instead to keep `HomeScreenContent` as `Stateless`, What do you think? 🙏 
```suggestion

```

Reply:
@Wadeewee You're right. However, the permission check composable needs to be separate from the `HomeScreenContent` and even `HomeScreen` composable to avoid re-composition.

Fixed in https://github.com/nimblehq/android-templates/pull/375/commits/decb5e2d24caf91d6a9ac742d4ea717c5b24f17c